src/common/snippets/src/lowered/expression.cpp

```diff
@@ -47,24 +46,24 @@ Expression::Expression(const std::shared_ptr<Node>& n,
 }
 
 const PortConnectorPtr& Expression::get_input_port_connector(size_t i) const {
-    assert(i < m_input_port_connectors.size() &&
-           "Failed to get input port connector: target input port must be less than input count!");
+    OPENVINO_DEBUG_ASSERT(i < m_input_port_connectors.size() &&
+                          "Failed to get input port connector: target input port must be less than input count!");
```

Comment:
Please double check for "&&" leftovers.
You can just grep for such pattern to make sure it is gone.